File: test/test_view/drugs/javascript.rb
#!/usr/bin/env ruby
# encoding: utf-8
# ODDB::View::Drugs::TestPrescription -- oddb.org -- 06.08.2012 -- yasaka@ywesee.com

$: << File.expand_path('../..', File.dirname(__FILE__))
$: << File.expand_path("../../../src", File.dirname(__FILE__))

gem 'minitest'
require 'minitest/autorun'

class TestJavaScript <Minitest::Test
  def test_simple_logging
    assert_equal("testing\n", `nodejs -e "console.log('testing');"`);
  end

  def test_prescription
    dir = File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..','src', 'view', 'drugs'))
    
    ENV['NODE_PATH']=dir
    expected = 'http://2dmedication.org/|1.0|4dd33f59-1fbb-4fc9-96f1-488e7175d761|TriaMed|3.9.3.0|7601000092786|K2345.33||20131104|Beispiel|Susanne|3073|19460801|7601003000382;|2014236||1||0.00-0.00-0.00-0.00|||1|||SPEZIALVERBAND|1|20131214|0.00-0.00-0.00-0.00||40|0|7680456740106|||1||1.00-1.00-1.00-0.00|zu Beginn der Mahlzeiten mit mindestens einem halben Glas Wasser einzunehmen||0;27834'
    assert_equal(expected, `nodejs #{__FILE__.sub('.rb','.js')}`.chomp);
  end
end

File: test/test_view/drugs/javascript.js
// Very crude unit test for the prescription java test
function create_qrcode_example() {
      var item1 = new PrescriptionItem();
      item1.pharmacode = '2014236';
      item1.may_be_substituted = false;

      var item2 = new PrescriptionItem
      item2.description = 'SPEZIALVERBAND'
      item2.valid_til = '20131214' // Time.new(2013, 12, 14)  // '20131214'
      item2.nr_repetitions = 40
      
      var item3 = new PrescriptionItem('7680456740106')
      item3.simple_posology = [1, 1.0, 1, 0]
      item3.extended_posology = 'zu Beginn der Mahlzeiten mit mindestens einem halben Glas Wasser einzunehmen'

      var prescription =  new create_example_prescription
      prescription.add_item(item1)
      prescription.add_item(item2)
      prescription.add_item(item3)
      return prescription
}

function create_example_prescription(){
      var prescription = new Prescription( '4dd33f59-1fbb-4fc9-96f1-488e7175d761');
      prescription.SW_ORIGIN_ID             = 'TriaMed'
      prescription.SW_VERSION_ID            = '3.9.3.0'
      prescription.doctor_glin              = '7601000092786'
      prescription.doctor_zsr               =  'K2345.33'
      prescription.patient_id               = ''
      prescription.date_issued              = '20131104' // Time.new(2013, 11, 04)
      prescription.patient_family_name      = 'Beispiel'
      prescription.patient_first_name       = 'Susanne'
      prescription.patient_zip_code         = '3073'
      prescription.patient_birthday         = '19460801' // Time.new(1946,8,1)
      prescription.patient_insurance_glin   = '7601003000382'
      return prescription   
}

console.log(create_qrcode_example().qr_string());
// TODO: add test for the following problematic string
// http://2dmedication.org/|1.0|62a5b6cb-43cd-84bc-a72e-98392d55ca41|ywesee GmBh|1.0|doctor-ean13|||2051|||||;7680516801112|||1||0.00-0.00-0.00-0.00|||0|7680576730063|||1||0.00-0.00-0.00-0.00|||0;14436
